l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py

```diff
@@ -0,0 +1,1556 @@
+            # 'origin': xmlData.datiOrdineAcquisto,
+            "fiscal_position_id": (partner.property_account_position_id.id or False),
+            "invoice_payment_term_id": partner.property_supplier_payment_term_id.id,
+            "company_id": company.id,
+            "fatturapa_attachment_in_id": fatturapa_attachment.id,
+            "narration": comment,
+        }
+
+        # 2.1.1.10
+        self.set_efatt_rounding(FatturaBody, invoice_data)
+
+        # 2.1.1.12
+        self.set_art73(FatturaBody, invoice_data)
+
+        # 2.1.1.5
+        wt_founds = self.set_withholding_tax(FatturaBody, invoice_data)
+
+        self.set_e_invoice_lines(FatturaBody, invoice_data)
+
+        invoice = invoice_model.create(invoice_data)
+
+        # 2.2.1
+        self.set_invoice_line_ids(
+            FatturaBody, credit_account_id, partner, wt_founds, invoice
+        )
+
+        # 2.1.1.7
+        self.set_welfares_fund(FatturaBody, credit_account_id, invoice, wt_founds)
+
+        invoice._onchange_invoice_line_wt_ids()
+        invoice._recompute_dynamic_lines()
+        invoice.write(invoice._convert_to_write(invoice._cache))
+
+        rel_docs_dict = {
+            # 2.1.2
+            "order": FatturaBody.DatiGenerali.DatiOrdineAcquisto,
+            # 2.1.3
+            "contract": FatturaBody.DatiGenerali.DatiContratto,
+            # 2.1.4
+            "agreement": FatturaBody.DatiGenerali.DatiConvenzione,
+            # 2.1.5
+            "reception": FatturaBody.DatiGenerali.DatiRicezione,
+            # 2.1.6
+            "invoice": FatturaBody.DatiGenerali.DatiFattureCollegate,
+        }
+
+        for rel_doc_key, rel_doc_data in rel_docs_dict.items():
+            if not rel_doc_data:
+                continue
+            for rel_doc in rel_doc_data:
+                doc_datas = self._prepareRelDocsLine(invoice.id, rel_doc, rel_doc_key)
+                for doc_data in doc_datas:
+                    # Note for v12: must take advantage of batch creation
+                    rel_docs_model.create(doc_data)
+
+        # 2.1.7
+        self.set_activity_progress(FatturaBody, invoice)
+
+        # 2.1.8
+        self.set_ddt_data(FatturaBody, invoice)
+
+        # 2.1.9
+        self.set_delivery_data(FatturaBody, invoice)
+
+        # 2.2.2
+        self.set_summary_data(FatturaBody, invoice)
+
+        # 2.1.10
+        self.set_parent_invoice_data(FatturaBody, invoice)
+
+        # 2.3
+        self.set_vehicles_data(FatturaBody, invoice)
+
+        # 2.4
+        self.set_payments_data(FatturaBody, invoice, partner_id)
+
+        # 2.5
+        self.set_attachments_data(FatturaBody, invoice)
+
+        self._addGlobalDiscount(
+            invoice.id, FatturaBody.DatiGenerali.DatiGeneraliDocumento
+        )
+
+        if self.e_invoice_detail_level != "1":
+            self.set_roundings(FatturaBody, invoice)
+
+        # compute the invoice
+        invoice._move_autocomplete_invoice_lines_values()
```

Comment:
@TheMule71 
In caso di fatture con arrotondamento IVA quel _move_autocomplete_invoice_line_values pare che faccia casino.
Ad esempio con quasi tutte le fatture amazon. Ho provato a capirci qualcosa ma non ho risolto nulla.
Posso fornire un XML di esempio.